src/main/java/mate/academy/bookstore/controller/OrderController.java

```diff
@@ -0,0 +1,83 @@
+package mate.academy.bookstore.controller;
+
+import io.swagger.v3.oas.annotations.Operation;
+import io.swagger.v3.oas.annotations.tags.Tag;
+import jakarta.validation.Valid;
+import java.util.List;
+import lombok.RequiredArgsConstructor;
+import mate.academy.bookstore.dto.request.order.CreateOrderRequestDto;
+import mate.academy.bookstore.dto.request.order.UpdateOrderRequestDto;
+import mate.academy.bookstore.dto.response.order.OrderItemResponseDto;
+import mate.academy.bookstore.dto.response.order.OrderResponseDto;
+import mate.academy.bookstore.entity.User;
+import mate.academy.bookstore.service.OrderService;
+import org.springframework.data.domain.Pageable;
+import org.springframework.security.access.prepost.PreAuthorize;
+import org.springframework.security.core.Authentication;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PatchMapping;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RestController;
+
+@Tag(name = "Orders management", description = "Endpoint for managing user orders")
+@RestController
+@RequiredArgsConstructor
+@RequestMapping("/api/orders")
+public class OrderController {
+    private final OrderService orderService;
+
+    @GetMapping
+    @PreAuthorize("hasRole('USER')")
+    @Operation(summary = "Get orders",
+            description = "Get a list of all user orders")
+    public List<OrderResponseDto> getOrders(Authentication auth, Pageable pageable) {
+        User user = (User) auth.getPrincipal();
+        return orderService.getAllOrders(user.getId(), pageable);
+    }
+
+    @GetMapping("/{orderId}/items")
+    @PreAuthorize("hasRole('USER')")
+    @Operation(summary = "Get order items",
+            description = "Get all items of a specific order by its id")
+    public List<OrderItemResponseDto> getOrderItems(@PathVariable Long orderId, Pageable pageable) {
+        return orderService.getOrderItems(orderId, pageable);

```

Comment:
PLease make sure that in all cases you work with order and order items of current user to avoid situation when one user can access order data of different user